apps/connect/src/store/document-model.ts

```diff
@@ -1,37 +1,37 @@
-import { module as DocumentDrive } from 'document-model-libs/document-drive';
-import { Action, DocumentModel } from 'document-model/document';
-import { module as DocumentModelModule } from 'document-model/document-model';
+import { useFeatureFlag } from '#hooks/useFeatureFlags/index';
+import { driveDocumentModelModule } from 'document-drive';
+import {
+    documentModelDocumentModelModule,
+    DocumentModelLib,
+    DocumentModelModule,
+} from 'document-model';
 import { atom, useAtomValue } from 'jotai';
 import { observe } from 'jotai-effect';
 import { atomWithLazy, unwrap } from 'jotai/utils';
-import { useFeatureFlag } from 'src/hooks/useFeatureFlags';
-import { DocumentModelsModule } from 'src/utils/types';
 import { atomStore } from '.';
 import { externalPackagesAtom } from './external-packages';
 
 export const LOCAL_DOCUMENT_MODELS = import.meta.env.LOCAL_DOCUMENT_MODELS;
 
 export const baseDocumentModels = [
-    DocumentDrive,
-    DocumentModelModule,
-] as const as DocumentModel[];
+    driveDocumentModelModule,
+    documentModelDocumentModelModule,
+] as DocumentModelModule<any, any>[];
 
 // removes document models with the same id, keeping the one that appears later
 function getUniqueDocumentModels(
-    ...documentModels: DocumentModel[][]
-): DocumentModel[] {
-    const uniqueModels = new Map<string, DocumentModel>();
-
-    for (const models of documentModels) {
-        for (const model of models) {
-            uniqueModels.set(model.documentModel.id, model);
-        }
+    ...documentModels: DocumentModelModule<any, any>[]
+): DocumentModelModule<any, any>[] {
+    const uniqueModels = new Map<string, DocumentModelModule<any, any>>();
+
+    for (const model of documentModels) {
+        uniqueModels.set(model.documentType, model);
     }
 
     return Array.from(uniqueModels.values());
 }
 
-function getDocumentModelsFromModules(modules: DocumentModelsModule[]) {
+function getDocumentModelsFromModules(modules: DocumentModelLib[]) {
```

Comment:
using the type goldenmule defined which is basically the same but has the manifest and generics